File: agcli/llm.py
from __future__ import annotations

from openai import OpenAI


def _normalize_model(model: str) -> str:
    # ag-style alias compatibility
    aliases = {
        "codex": "gpt-5.3-codex",
        "quality": "gpt-5.3-codex",
        "fast": "gpt-5-mini",
    }
    return aliases.get(model, model)


def ask_model(api_key: str, model: str, prompt: str, endpoint: str | None = None) -> str:
    client = OpenAI(api_key=api_key, base_url=endpoint) if endpoint else OpenAI(api_key=api_key)
    real_model = _normalize_model(model)
    res = client.responses.create(
        model=real_model,
        input=prompt,
    )
    return res.output_text

File: src/llm.js
import OpenAI from 'openai';

function normalizeModel(model) {
  const aliases = {
    codex: 'gpt-5.3-codex',
    quality: 'gpt-5.3-codex',
    fast: 'gpt-5-mini'
  };
  return aliases[model] || model;
}

export async function askModel({ apiKey, model, prompt, endpoint }) {
  const client = new OpenAI({ apiKey, baseURL: endpoint || undefined });
  const realModel = normalizeModel(model);
  const res = await client.responses.create({
    model: realModel,
    input: prompt
  });
  return res.output_text || '';
}
